data_model/src/lib.rs

```diff
@@ -532,6 +532,8 @@ pub enum Value {
     Ipv4Addr(iroha_primitives::addr::Ipv4Addr),
     /// IP Version 6 address.
     Ipv6Addr(iroha_primitives::addr::Ipv6Addr),
+    /// AssetValue
+    AssetValue(AssetValue),
```

Comment:
now we seem to have some redundancy here because the variants of the `AssetValue` are also variants of the `Value`

Reply:
Agree with you, maybe we can introduce something like this? 

```rust
enum NumericValue {
   U32(u32),
   U128(u128),
   Fixed(Fixed),
}

enum Value  {
    // ...
    Numeric(NumericValue),
    // ...
    // delete U32, U128 and Fixed variants
}

// we can still accept `u32`, `u128` and `Fixed` by implementing `TryFrom<T> for Value`.

enum AssetValue {
     Numeric(NumericValue),
     Store(Metadata),
}
```